### PR TITLE
Add RDataFrame to the primer

### PR DESCRIPTION
**Pull request by vepadulano**
This commit adds RDataFrame content to the ROOT primer. It also removes obsolete content about TNTuple, TSelector and PROOF Lite to discuss data analysis in terms of RDataFrame. Parts of the old content that were referring to the underlying data format have been kept and reworded under a new section about the ROOT dataset

Co-authored-by: Ivan Kabadzhov <kabadzhovivan@gmail.com>
Co-authored-by: Enrico Guiraud <enrico.guiraud@cern.ch>

### CODE REVIEW

**Review by eguiraud — CHANGES_REQUESTED** [review]
```
build/724/primer/index.html:3417:<a href="https://root.cern.ch/doc/master/classROOT_1_1RDataFrame.html#parallel-execution">in the RDataFrame documentation</a>.</p>
build/724/primer/index.html:3430:<a href="https://root.cern.ch/doc/master/classROOT_1_1RDataFrame.html#python">in the RDataFrame documentation</a>.</p>

Found 2 links to root.cern.ch. Please change them to link to root.cern (no '.ch') instead.
```

**Comment by couet** [comment]
I see the NTuple paragraph showing how to use TNtuple has been completely remove. I understand we want to remove the PROOF part (as PROOF is now old software and has been declared as such). But TNtuple has not been marked as legacy code and many tutorials are using it. Removing it from the primer is a "political" (I do not have better word) choice, which is perfectly understandable. I think it is fair this choice if done by the new young generation of the ROOT developers and not by an old developper like me. So I approve this PR.

**Review by couet — APPROVED** [review]

**Comment by couet** [comment]
Also, the [original markdown version of the primer](https://github.com/root-project/root/tree/master/documentation/primer) is still in the ROOT repository. It allows to generate printable version of the primer (pdf, epub ...). With this PR this markdown primer will not be in sync with the html one in the ROOT web site. My question is: Should we completely remove this markdown primer for the ROOT repo ?

**Comment by couet** [comment]
One more comment:
At the top of the html Primer we have:

_Original Authors: D. Piparo, G. Quast, M. Zeise_

Given the major changes this PR provides, I guess it is worth mentioning the new authors (Ivan Enrico Vincenzo)

**Comment by eguiraud** [comment]
You tell us :grimacing:

**Comment by couet** [comment]
> You tell us 
I made two PRs to remove the Markdown version

And, yes, I will be in favour to remove the "interactive one" also (need to check first from where it is accessible)

File: primer/index.md [hunk 1]
```diff
@@ -2024,561 +2025,194 @@ void read_from_file(){
...
     auto myCanvas = new TCanvas();
     h->DrawClone();
 }
 {% endhighlight %}
 
-## N-tuples in ROOT
-
-***Storing simple N-tuples***
+## The ROOT dataset
+ROOT offers its own data format with the TTree class. These are some of its advantages:
```
Comment (jalopezg-r00t) [inline primer/index.md:2037]: Given that the ROOT file format is independent from TTree, I would suggest to rephrase this as:
```suggestion
ROOT can handle storage/retrieval of large datasets through the {% include ref class="TTree" %} class. These are some of its advantages:
```

File: primer/index.md [hunk 2]
```diff
@@ -2024,561 +2025,194 @@ void read_from_file(){
...
+  measurement should be stored in a separate tree. Although the
+  compression can handle redundant values, the processing time
+  increases with every variable that has to be filled.
+- The function SetCacheSize(long) specifies the size of the cache
+  for reading a TTree object from a file. The default value is 30MB.
+  A manual increase may help in certain situations. Please note that
+  the caching mechanism can cover only one TTree object per TFile
+  object.
+- If you would like to measure I/O performance when reading a dataset you could
+  use the TTreePerfStats class. The ROOT documentation on this class also
```
Comment (jalopezg-r00t) [inline primer/index.md:2065]:
```suggestion
  use the {% include ref class="TTreePerfStats" %} class. The ROOT documentation on this class also
```

File: primer/index.md [hunk 3]
```diff
@@ -2024,561 +2025,194 @@ void read_from_file(){
...
-entry to be processed. A template for the class `TSelector` is provided
-by the method `TTree::MakeSelector`, as is shown in the little macro
-`makeSelector.C` below.
+## Working with columns that hold collections
 
-It opens the n-tuple `conductivity_experiment.root` from the example
-above and creates from it the header file `MySelector.h` and a template
-to insert your own analysis code, `MySelector.C`.
+RDataFrame reads collections as the special type
+[ROOT::RVec](https://root.cern/doc/master/classROOT_1_1VecOps_1_1RVec.html):
```
Comment (jalopezg-r00t) [inline primer/index.md:2168]:
```suggestion
[ROOT::VecOps::RVec](https://root.cern/doc/master/classROOT_1_1VecOps_1_1RVec.html):
```

File: primer/index.md [hunk 4]
```diff
@@ -2024,561 +2025,194 @@ void read_from_file(){
...
+for example, a column where each element is an array of floating point numbers
+can be read as a ROOT::RVecF. C-style arrays (with variable or static size),
```
Comment (jalopezg-r00t) [inline primer/index.md:2170]: Given that we talked about the `RVec` type (and not `RVecF`), I would prefer to stick to that, i.e.
```suggestion
can be read as a `ROOT::VecOps::RVec<float>`. C-style arrays (with variable or static size),
```

File: primer/index.md [hunk 5]
```diff
@@ -2024,561 +2025,194 @@ void read_from_file(){
...
+STL vectors and most other collection types can be read this way.
 
+RVec is a container similar to std::vector (and can be used just like a
+std::vector) but it also offers a rich interface to operate on the array
```
Comment (jalopezg-r00t) [inline primer/index.md:2174]:
```suggestion
`RVec` is a container similar to `std::vector` (and can be used just like a
`std::vector`) but it also offers a rich interface to operate on the array
```

File: primer/index.md [hunk 6]
```diff
@@ -2024,561 +2025,194 @@ void read_from_file(){
...
-
-The entry points `SlaveBegin()` and `SlaveTerminate()` are called on
-so-called slave nodes only if parallel processing via `PROOF` or
-`PROOF lite` is enabled, as will be explained below.
-
-A simple example of a selector class is shown in the macro
-`MySelector.C`. The example is executed with the following sequence of
-commands:
+For example, to fill a histogram with the "pt" of selected particles for each
+event, Define() can be used to create a column that contains the desired array
```
Comment (jalopezg-r00t) [inline primer/index.md:2178]:
```suggestion
event, `Define()` can be used to create a column that contains the desired array
```

File: primer/index.md [hunk 7]
```diff
@@ -2024,561 +2025,194 @@ void read_from_file(){
...
-Info in <TPacketizerAdaptive::TPacketizerAdaptive>:
-                      Setting max number of workers per node to 4
-Validating files: OK (4 files)
-Info in <TPacketizerAdaptive::InitStats>:
-                      fraction of remote files 1.000000
-Info in <TCanvas::Print>:
-       file ResistanceDistribution.png has been created
-*==* ----- End of Job ----- Date/Time = Wed Feb 15 23:00:08 2012
-Lite-0: all output objects have been merged
+{% highlight Python %}
```
Comment (jalopezg-r00t) [inline primer/index.md:2208]:
```suggestion
{% highlight Python %}
import ROOT
```

**Review by jalopezg-r00t — COMMENTED** [review]
LGTM; just some minor comments.

File: primer/index.md (hunk 3 above)
Comment (eguiraud) [reply primer/index.md:2168]: I'd rather advertise it as `ROOT::RVec` here (it's valid)

File: primer/index.md (hunk 4 above)
Comment (eguiraud) [reply primer/index.md:2170]: we should advertise the short aliases though

**Comment by Axel-Naumann** [comment]
Do we have a chance to keep a notebook version of the primer? Or do yous think that it's not worth it?

**Comment by couet** [comment]
I think it will be much simpler to maintain only one version of the Primer. We have plenty of tutorials in Notebook format. The Primer is really a manual. I am not sure it makes sense to have it as a Notebook. Moreover, this PR highlight the fact the Primer is something important to have, it evolves, it is not frozen. Several versions will be a nightmare to maintain. I would go toward simplification.

**Comment by Axel-Naumann** [comment]
> The Primer is really a manual.

How so - it seems to have lot of code?

My question isn't whether we can have duplication but whether we should drop the notebook. I.e. can we either generate the notebook from the single source, or should the notebook be the single source?

**Comment by couet** [comment]
> How so - it seems to have lot of code?

Yes but in the "Manual" and in the "reference Guide"there is a lot of code too.

I must admit I do not really know how the Notebook is generated. I need to check. I do not even know where it can seen. I try to find it online yesterday without success.

File: primer/index.md (hunk 4 above)
Comment (vepadulano) [reply primer/index.md:2170]: Maybe I can write a short sentence describing the existence of the aliases

Comment (vepadulano) [reply primer/index.md]: I've added a small sentence there. If it looks ok this can be merged

Comment (eguiraud) [reply primer/index.md]: I'd like to figure out what to do with the other versions of the primer before merging
